src/audit_cmd.py

```diff
@@ -20,42 +20,59 @@ def take_action(self, parsed_args):
         log_group_blacklist_svc = LogGroupsBlacklistService(self.app.config['dynamodb'])
 
         # Load data
-        self.log.info('Fetching log groups ...')
-        log_groups = log_group_svc.list()
-
         self.log.info('Fetching environments ...')
         environments = EnvironmentsService(self.app.config['dynamodb']).list()
 
         self.log.info('Fetching log ingestion streams ...')
         ingestion_streams = LogIngestionStreamsService(self.app.config['dynamodb']).list()
-
-        self.log.info('Fetching blacklisted log groups ...')
-        blacklisted_log_groups = log_group_blacklist_svc.list()
-
         ingestion_streams_indexed = {}
         for ingestion_stream in ingestion_streams:
             ingestion_streams_indexed[ingestion_stream['environment_id']] = ingestion_stream
 
+        self.log.info('Fetching blacklisted log groups ...')
+        blacklisted_log_group_arns = log_group_blacklist_svc.list()
+
+        self.log.info('Fetching log groups ...')
+        log_groups = log_group_svc.list()
+        log_groups = self.non_blacklisted_log_groups(log_groups, blacklisted_log_group_arns)
+
+        self.log.info('Fetching log group subscriptions ...')
+        log_group_subscriptions = {}
+        for log_group in log_groups:
+            log_group_name = log_group['logGroupName']
+            subscription = log_group_svc.get_subscription_filters(log_group_name)
+            log_group_subscriptions[log_group_name] = subscription
+        log_groups = self.non_subscribed_log_groups(log_groups, log_group_subscriptions)
+
+        self.log.info('Fetching log group tags ...')
+        log_group_tags = {}
+        for log_group in log_groups:
+            log_group_name = log_group['logGroupName']
+            tags = log_group_svc.get_tags(log_group_name)
+            log_group_tags[log_group_name] = tags
+
         # Subscription options
         options = self.generate_options(environments)
 
+        # Do audit
         for log_group in log_groups:
-            if len(log_group.get('subscriptionFilters')) == 0 and log_group['arn'] not in blacklisted_log_groups:
-                option = self.prompt_for_sub_environment(log_group, options)
-                if not option:
-                    # User asked to exit
-                    return
-
-                sub_environment_id = option['environment_id']
-                if sub_environment_id:
-                    ingestion_stream = ingestion_streams_indexed[sub_environment_id]
-                    log_group_svc.subscribe_to_kinesis(
-                        log_group['logGroupName'],
-                        ingestion_stream['destination_arn'],
-                        ingestion_stream['iam_role_arn']
-                    )
-                else:
-                    log_group_blacklist_svc.create(log_group['arn'])
+            log_group_name = log_group['logGroupName']
+            log_group['tags'] = log_group_tags[log_group_name]
```

Comment:
?

Reply:
The tags are used in the `prompt_for_sub_environment` function